src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioManagedModuleUpdateProvider.cs

```diff
@@ -28,31 +29,18 @@ public VisualStudioManagedModuleUpdateProvider(VisualStudioWorkspace workspace)
             _encService = workspace.Services.GetRequiredService<IEditAndContinueWorkspaceService>();
         }
 
-        public Task<ManagedModuleUpdateStatus> GetStatusAsync(CancellationToken cancellationToken)
-            => GetStatusAsync(null, cancellationToken);
-
         /// <summary>
-        /// Returns the state of the changes made to the source. 
-        /// The EnC manager calls this to determine whether there are any changes to the source 
-        /// and if so whether there are any rude edits.
-        /// 
-        /// TODO: Future work in the debugger https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1051385 will replace this with bool HasChangesAsync.
-        /// The debugger currently uses <see cref="SolutionUpdateStatus.Ready"/> as a signal to trigger emit of updates 
-        /// (i.e. to call <see cref="GetManagedModuleUpdatesAsync(CancellationToken)"/>). 
-        /// When <see cref="SolutionUpdateStatus.Blocked"/> is returned updates are not emitted.
-        /// Since <see cref="GetManagedModuleUpdatesAsync(CancellationToken)"/> already handles all validation and error reporting 
-        /// we either return <see cref="SolutionUpdateStatus.None"/> if there are no changes or <see cref="SolutionUpdateStatus.Ready"/> if there are any changes.
+        /// Returns true if any changes have been made to the source since the last changes had been applied.
         /// </summary>
-        public async Task<ManagedModuleUpdateStatus> GetStatusAsync(string sourceFilePath, CancellationToken cancellationToken)
+        public async Task<bool> HasChangesAsync(string sourceFilePath, CancellationToken cancellationToken)
```

Comment:
It's odd that this method returns true for exceptions as well. Maybe naming can help clarify. `IsBlockedAsync`?

Reply:
It implements debugger interface thru which the debugger is asking if there are any changes to be applied. 
If we have a bug (an exception is thrown) in our impl we just need to reasonably recover. It's better to assume there are changes so that the debugger tries to apply them, which reports more details to the user (such as rude edits).